utilities/checkpoint/checkpoint_impl.cc

```diff
@@ -58,13 +58,22 @@ Status CheckpointImpl::CleanStagingDirectory(
                  s.ToString().c_str());
 
   s = db_->GetEnv()->GetChildren(full_private_path, &subchildren);
+  if (s.IsNotFound()) {
```

Comment:
Why would this return NotFound() if FileExists() is not NotFound above? Do we have concurrent CleanStagingDirectory() with the same path?

Reply:
Oh, should we return early if FileExists() returns a non-ok status above?

Reply:
> Oh, should we return early if FileExists() returns a non-ok status above?

Oh good catch. Let me fix this. 

> Why would this return NotFound() if FileExists() is not NotFound above? Do we have concurrent CleanStagingDirectory() with the same path?

Yeah this could happen by observation. " Do we have concurrent CleanStagingDirectory() with the same path?" not sure about this part. But let me check.

Reply:
I don't think we create checkpoint at the same directory concurrently. It's probably because the FileExists() call above failed with some injected error.

Reply:
Right - I overlooked the part where we didn't early return. Will fix it.

Reply:
fixed